feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/PreviewViewModel.kt

```diff
@@ -84,6 +98,22 @@ class PreviewViewModel @Inject constructor(
         }
     }
 
+    fun setFlash(flashModeStatus: FlashModeStatus){
+        //update viewmodel value
+        viewModelScope.launch {
+            _previewUiState.emit(
+                previewUiState.value.copy(
+                    currentCameraSettings =
+                        previewUiState.value.currentCameraSettings.copy(
+                            flash_mode_status = flashModeStatus
+                        )
+                )
+            )
+        }
+        // apply to cameraUseCase
+        cameraUseCase.setFlashMode(previewUiState.value.currentCameraSettings.flash_mode_status)
```

Comment:
Since the flash mode is being read from the preview ui state, it's possible to be reading a stale state.
The `launch` operation may not trigger immediately since it's starting a coroutine. So that means the coroutine has to launch and then the preview ui state has to be updated with the new value. But this line might be called before all that happens and so the value being read here might be the previous value.

One option is to invoke the `cameraUseCase.setFlashMode(flashModeStatus)` first and then call viewModeScope.launch
Or perform everything inside the viewmodelscope